### ISSUE TITLE
Hit region (Canvas 2D) support

### ISSUE THREAD

**Issue by cmdrflexo** [webmixedreality/exokit#763]
Canvas 2D [hit regions](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/addHitRegion#Browser_compatibility) are experimental and required to be explicitly enabled on Chrome.

![image](https://user-images.githubusercontent.com/4148613/51784725-7c9cdf80-211b-11e9-9a80-e62ad357de7c.png)

Hit regions result in much easier click-detection for 2D canvas elements as well as more efficient and readable code.

```js
context.rect(10,10,100,100)
context.fill()
context.addHitRegion({"id": "Hit Region"})

canvas.onclick = event => {
    if(event.region) alert("You clicked " + event.region)
}
```

I highly recommend having a discussion about this feature.

**Comment by avaer** [comment]
Since exploit doesn’t draw canvas2d except as a texture, and there is no 3D hit test against textures, how would this fit in?

**Comment by avaer** [comment]
Since exokit doesn’t draw canvas2d except as a texture, and there is no 3D hit test against textures, how would this fit in?

**Comment by cmdrflexo** [comment]
The canvas still exists, but yeah, right now everything that could support this is tossed out. I don't know enough about the technicals to give you a great answer, but my suggestion is to keep it in mind if at all possible. Any application that wants to draw to a 2D surface, for UI for example, runs into hurdles with exokit being so 3D-centric. 2D rendering for interactive elements shouldn't be disregarded. Current support isn't terrible for 2D rendering, with canvas-to-texture tricks, but a couple of prime features, such as hit regions, could result in a slim, concise framework to pull of a lot what's required when it comes to interactive interfaces.

**Comment by aoighost** [comment]
I agree with flexo. Part of what I want to do is be able to have VNC pull up in a window in XR. Being able to have hit regions and things like that would help dramatically.
